plugins/pulp_python/plugins/importers/importer.py

```diff
@@ -183,7 +185,11 @@ def upload_unit(self, repo, type_id, unit_key, metadata, file_path, conduit, con
         :rtype:           dict
         """
         package = models.Package.from_archive(file_path)
-        package.save_and_import_content(file_path)
+        try:
+            package.save_and_import_content(file_path)
+        except NotUniqueError:
+            package = package.__class__.objects.filter(**package.unit_key).first()
```

Comment:
You can also use .get() instead of .filter().first() which will get you one object. However, get() does raise an Exception if nothing matches.